scripts/missions/amk/09_Rescue_A_Moogles_Labor_of_Love.lua

```diff
@@ -0,0 +1,208 @@
+-----------------------------------
+-- Rescue! A Moogle's Labor of Love
+-- A Moogle Kupo d'Etat M9
+-- !addmission 10 8
+-- Geologist cutscene args : csid, progress, has QC map: 1 or 0, markerset: 1-10
+-- Goblin Geologist        : !pos -737 -6 -550 208
+-- STONE_OF_SURYA          : !addkeyitem 1145
+-- STONE_OF_CHANDRA        : !addkeyitem 1146
+-- STONE_OF_MANGALA        : !addkeyitem 1147
+-- STONE_OF_BUDHA          : !addkeyitem 1148
+-- STONE_OF_BRIHASPATI     : !addkeyitem 1149
+-- STONE_OF_SHUKRA         : !addkeyitem 1150
+-- STONE_OF_SHANI          : !addkeyitem 1151
+-- STONE_OF_RAHU           : !addkeyitem 1152
+-- STONE_OF_KETU           : !addkeyitem 1153
+-- NAVARATNA_TALISMAN      : !addkeyitem 1158
+-----------------------------------
+require('scripts/globals/missions')
+require('scripts/globals/interaction/mission')
+local ID = zones[xi.zone.QUICKSAND_CAVES]
+-----------------------------------
+
+local mission = Mission:new(xi.mission.log_id.AMK, xi.mission.id.amk.RESCUE_A_MOOGLES_LABOR_OF_LOVE)
+
+mission.reward =
+{
+    nextMission = { xi.mission.log_id.AMK, xi.mission.id.amk.ROAR_A_CAT_BURGLAR_BARES_HER_FANGS },
+}
+
+local markerSets =
+{
+    { 1, 2, 5, 7,  8,  11, 14, 19, 20 },
+    { 3, 4, 6, 9,  11, 13, 16, 17, 18 },
+    { 2, 6, 7, 8,  10, 12, 15, 19, 20 },
+    { 1, 3, 4, 5,  8,  9,  10, 17, 18 },
+    { 2, 4, 7, 11, 12, 13, 15, 16, 20 },
+    { 1, 3, 5, 6,  8,  9,  14, 18, 19 },
+    { 2, 5, 7, 10, 11, 12, 15, 16, 17 },
+    { 1, 3, 4, 6,  8,  13, 14, 17, 20 },
+    { 2, 4, 7, 9,  10, 11, 16, 18, 19 },
+    { 3, 5, 6, 12, 13, 14, 15, 18, 20 },
+}
+
+local getMarkerSet = function(player)
+    -- markerSet is the setIndex of a random table within markerSets defined above
+    local markerSet = player:getCharVar('Mission[10][8]markerSet')
+    if markerSet == 0 then
+        markerSet = math.random(1, #markerSets)
+        player:setCharVar('Mission[10][8]markerSet', markerSet)
+    end
+
+    return markerSet
+end
+
+local hasAllStones = function(player)
+    for offset = 0, 8 do
```

Comment:
The loop would probably be more clear instead of referring to offset, but iterating `for keyItemId = xi.ki.STONE_OF_SURYA, xi.ki.<whatever_is_end>`